.travis.yml

```diff
@@ -30,17 +30,39 @@ linux_gcc: &linux_gcc
       sources: [ 'ubuntu-toolchain-r-test' ]
       packages: [ 'g++-4.9' ]
 
+mpi_linux_clang: &mpi_linux_clang
+  os: linux
+  compiler: "clang-3.8"
+  addons: 
+    apt:
+      sources: [ 'ubuntu-toolchain-r-test', 'llvm-toolchain-precise-3.8' ]
+      packages: [ 'clang-3.8', 'libc++-dev', 'mpich', 'libmpich-dev' ]
+
+mpi_linux_gcc: &mpi_linux_gcc
+  os: linux
+  compiler: "gcc-4.9"
+  addons:
+    apt:
+      sources: [ 'ubuntu-toolchain-r-test' ]
+      packages: [ 'g++-4.9', 'mpich', 'libmpich-dev' ]
+
 matrix:
   fast_finish: true
   include:
     - <<: *osx_clang
       env:
         - MATRIX_EVAL="CXX=\"clang++ -DSTAN_THREADS\""
           TESTFOLDER="test/unit/math/rev/core test/unit/math/rev/mat/functor"      
-    - <<: *linux_clang
+    - <<: *mpi_linux_clang
       env:
-        - MATRIX_EVAL="CXX=clang++-3.8"
-          TESTFOLDER="test/unit/math/memory test/unit/math/prim/scal test/unit/math/prim/arr"
```

Comment:
Did you mean to get rid of the memory, prim/scal, and prim/arr tests for linux_clang?

Reply:
Oh, NO! Let me double check this (this PR is to increase test-converage).

Reply:
Looks like I changed the lines towards what I wanted them without copying them first. Fixed now. Thanks for catching.